### ISSUE TITLE
Guideline id does not seem to be supported when converting between path/values and compositions

### ISSUE THREAD

**Issue by ralphvanetten** [ethercis/ethercis#6]
I used the `Generic Laboratory Test Report.v0` from the CMK to test this (also attached) :

The most basic unit test I have which fails because the guideline ID is not set:
```
final Map<String, Object> values = new HashMap<>();
values.put(
    "[openEHR-EHR-COMPOSITION.report-result.v1]/content[openEHR-EHR-OBSERVATION.laboratory_test.v0]/guideline_id", 
    "test"
);

final PathValue pathValue = new PathValue(
    knowledgeManager,
    "Generic Laboratory Test Report.v0", 
    null
);

final Composition composition = pathValue.assign(values);

assertNotNull(((Observation)composition.getContent().get(0)).getGuidelineId());
```

If I manually set the guideline ID and then convert it back to path/values, it throws the following exception: `java.lang.ClassNotFoundException: com.ethercis.ehr.encode.wrappers.ObjectRefVBean` in `com.ethercis.ehr.util.MapInspector.generateObject(MapInspector.java:284)`

```
final Map<String, Object> values = new HashMap<>();
values.put(
    "[openEHR-EHR-COMPOSITION.report-result.v1]/content[openEHR-EHR-OBSERVATION.laboratory_test.v0]/guideline_id", 
    "test"
);

final PathValue pathValue = new PathValue(
    knowledgeManager,
    "Generic Laboratory Test Report.v0", 
    null
);

final Composition composition = pathValue.assign(values);

// manually set guideline ID
((Observation)composition.getContent().get(0)).setGuidelineId(
    new ObjectRef(new GenericID("test", "s"), "ns", "t")
);

// this throws the exception
final Map<String, String> data = new EcisFlattener().render(composition);
```

**Comment by ethercis** [comment]
Yes you are right, care_entry attributes are not supported at this stage. The reason is that, to date, we didn't have any use case for their usage (@freshehr do you know any usage of these?)

**Comment by freshehr** [comment]
I have not come across use of guideline_id so far. Having said that, it is optional and is not used by the generic lab report template, so should not cause a fail.

**Comment by ralphvanetten** [comment]
For our use case we are using guideline_id to identify the type of care plan to ensure only a single care plan of a certain type is open at the same time. 

Since nobody seems to be using guideline_id how are others keeping track of the care plan type? Are we misunderstanding the purpose of the guideline_id? I can see a few possibilities such as using a coded text for the care plan name or using specialized archetypes. What is the best solution?

The ultimate goal is is to ensure someone can not add for example a new asthma care plan if there already is an active asthma care plan.

Thanks.

**Comment by ralphvanetten** [comment]
After some discussion it seems guideline_id is supposed to be used for other things. The suggested solution is to add the type of care plan as a normal field to the archetypes so we will do just that.

As it seems guideline_id is unused this issue can be of really low priority.
